client-src/elements/form-field-specs.js

```diff
@@ -2035,3 +2036,25 @@ async function checkFirstEnterpriseNotice(value, initialValue) {
   }
   return undefined;
 }
+
+async function checkExtensionMilestoneIsValid(value) {
```

Comment:
Would it make sense to require that the extension milestone is greater than the previous OT ending milestone?

Reply:
I believe this is a good idea, but after playing with things, it seemed deceptively difficult to make sure the extension milestone was before the OT milestone in the semantic checks. I've added a TODO here to cover this, but this might take a deeper dive into understanding how the semantic checks work to get this set up.